R/names.R

```diff
@@ -0,0 +1,405 @@
+#' Retrieve and repair names
+#'
+#' @description
+#'
+#' Like `base::names()`, `vec_names()` retrieves the names of a
+#' vector. It has the following differences:
+#'
+#' * It always returns __minimal names__. Whereas `names()` returns
+#'   `NULL` when a vector doesn't have names, `vec_names()` returns a
+#'   character vector of empty strings by default. `NA` names are also
+#'   converted to `""`. You can control how names are repaired with
+#'   the `repair` argument.
+#'
+#' * With arrays it returns the row names. With data frames it returns
+#'   an empty character vector as long as the number of rows.
```

Comment:
This data frame behaviour is very unexpected to me.

Reply:
We discussed with @hadley about returning the row names instead.

Reply:
I wonder if we need to (also) expose the name repair as a function that takes a character vector.

Reply:
@jennybc I think the invariant here is that `vec_size(vec_names(x))` needs to equal `vec_size(x)`. But we need to figure out how to resolve that with the naming we need for data frames etc.